packages/pigeon/test/java_generator_test.dart

```diff
@@ -818,9 +818,10 @@ void main() {
     expect(code, contains('Long add(Long x, Long y)'));
     expect(
         code, contains('ArrayList<Object> args = (ArrayList<Object>)message;'));
-    expect(code, contains('Long xArg = (Long)args.get(0)'));
-    expect(code, contains('Long yArg = (Long)args.get(1)'));
-    expect(code, contains('Long output = api.add(xArg, yArg)'));
+    expect(code, contains('Number xArg = (Number)args.get(0)'));
+    expect(code, contains('Number yArg = (Number)args.get(1)'));
+    expect(code,
+        contains('Long output = api.add(xArg.longValue(), yArg.longValue())'));
```

Comment:
Why break the conversion to Long into two separate pieces? I.e., why not:
```
Long xArg = ((Number)args.get(0)).longValue();
...
... api.add(xArg, yArg);
```

It seems odd to construct an intermediate that's explicitly intended to be the argument, but give it the wrong type at that stage.

Reply:
Yea, it was just a quirk where it was less code on the generator to do it this way.  This way I don't need to have differing types in the right hand and left hand expressions or selectively add parenthesis.